### PR TITLE
To use the list jobs in list_materialization_operation

### PR DESCRIPTION
**Pull request by anyalee0221**
# Description

Change the implementation of list_materialization_operation to jobs operation list which supported in 08-01 version

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_feature_set/feature_set_materialization_metadata.py [hunk 1]
```diff
@@ -45,17 +45,30 @@ def __init__(
...
-            tags=obj.tags,
+            type=FeaturestoreJobTypeMap.get(job_type),
+            feature_window_start_time=feature_window_start_time,
+            feature_window_end_time=feature_window_end_time,
+            name=obj.name,
+            display_name=job_properties.display_name,
+            creation_context=SystemData(created_at=obj.system_data.created_at),
+            status=job_properties.status,
+            tags=job_properties.tags,
+            duration=None,
```
Comment (shail2208) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_feature_set/feature_set_materialization_metadata.py:73]: Do we not have duration in the JobBase?

Comment (anyalee0221) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_feature_set/feature_set_materialization_metadata.py:73]: Yes, there's no duration found in JobBase.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_feature_set/feature_set_materialization_metadata.py [hunk 2]
```diff
@@ -69,17 +69,30 @@ def __init__(
         self.tags = tags
 
     @classmethod
-    def _from_rest_object(cls, obj: RestFeaturesetJob) -> "FeatureSetMaterializationMetadata":
+    def _from_rest_object(cls, obj: RestJobBase) -> "FeatureSetMaterializationMetadata":
```
Comment (shail2208) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_feature_set/feature_set_materialization_metadata.py:72]: Just make sure that We don't use _from_rest_object for RestFeaturesetJob anywhere else in the code.

Comment (anyalee0221) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_feature_set/feature_set_materialization_metadata.py:72]: Did a repo wide search, the RestFeaturesetJob  only apears in this file, and it's updated to RestJobBase.

**Review by fredms — APPROVED** [review]

**Review by qjxu — APPROVED** [review]

**Review by shail2208 — APPROVED** [review]

**Pull request merged**